### ISSUE TITLE
Save parameters used by CliMA in output

### ISSUE THREAD

**Issue by trontrytel** [CliMA/ClimateMachine.jl#1649]
### Description

Would it be useful to automatically dump all the parameters actually used by `ClimateMachine.jl` into output? 

Right now we overwrite the default `CliMAParameters.jl` in the experiment setup files (like for example here: https://github.com/CliMA/ClimateMachine.jl/pull/1564/files#diff-bbbc555ecaca9e76512a1f7f71b1d6d1eddb789644c982507113f927f4eca0daR45-R53, thank you @charleskawczynski ).
And with this approach, looking at the `CLIMAParameters.jl` branch used in the simulation will not give the correct answer. 

It's not a problem now. But as more people start experimenting with the model and changing their default parameters, it may become useful/needed. @kpamnany , @charleskawczynski  - Any thoughts?

**Comment by tapios** [comment]
Yes! Good point. We should have log files collecting all important settings for a run, including parameters.

**Comment by kpamnany** [comment]
Agree also. I was imagining something like this working bi-directionally so that you could ask to use the settings from a log for a new run.
